webapp/app/plugins/controller.js

```diff
@@ -235,25 +231,43 @@ treeherder.controller('PluginCtrl', [
          * Get the build_id needed to cancel or retrigger from the currently
          * selected job.
          */
-        var getRequestId = function() {
+        var getBuildbotRequestId = function() {
             if ($scope.artifacts.buildapi) {
                 return $scope.artifacts.buildapi.blob.request_id;
-            } else {
-                // this is super unlikely since we'd need to have at least one of those
-                // artifacts to even create the job in treeherder.  This is just a fallback...
-                thNotify.send("Unable to get request id for retrigger/cancel", "danger", true);
-                return null;
             }
         };
 
         $scope.retriggerJob = function() {
-            thBuildApi.retriggerJob($scope.repoName, getRequestId());
+            ThJobModel.retrigger($scope.repoName, $scope.job.id).then(function() {
+                var requestId = getBuildbotRequestId();
+                if (requestId) {
+                    return thBuildApi.retriggerJob($scope.repoName, requestId);
```

Comment:
I'm not quite clear on why you call ``ThJobModel.retribber`` and then also call ``thBuildApi.retriggerJob``.  Wouldn't that do a retrigger for on buildbot jobs twice?